frontend/src/slices/orders/order.slice.ts

```diff
@@ -4,18 +4,27 @@ import { DataStatus } from '~/libs/enums/enums.js';
 import { type ValueOf } from '~/libs/types/types.js';
 import { type OrderResponseDto } from '~/packages/orders/orders.js';
 
-import { calculateOrderPrice, createOrder } from './actions.js';
+import {
+  calculateOrderPrice,
+  createOrder,
+  getOrder,
+  getRouteData,
+  updateOrderFromSocket,
+} from './actions.js';
+import { type RouteData } from './libs/types/route-data.type.js';
 
 type State = {
-  orders: OrderResponseDto[];
+  orders: (OrderResponseDto | undefined)[];
```

Comment:
Can you tell me in what cases it will be undefined? Just wondering.

Reply:
At some point eslint not accepted it without this type (it was something like no-unnecessary-condition.), but now it works without it, good point